backend/kernelCI_app/tests/unitTests/helpers/filters_helpers_test.py

```diff
@@ -0,0 +1,1355 @@
+        """Test is_status_failure with custom fail list."""
+        custom_fail_list = ["CUSTOM_FAIL", "ERROR"]
+        result = is_status_failure("CUSTOM_FAIL", custom_fail_list)
+        assert result is True
+
+    def test_is_status_failure_with_custom_fail_list_no_match(self):
+        """Test is_status_failure with custom fail list no match."""
+        custom_fail_list = ["CUSTOM_FAIL", "ERROR"]
+        result = is_status_failure("PASS", custom_fail_list)
+        assert result is False
+
+
+class TestIsKnownIssue:
+    def test_is_known_issue_with_valid_issue(self):
+        """Test is_known_issue with valid issue."""
+        result = is_known_issue("issue123", 1)
+        assert result is True
+
+    def test_is_known_issue_with_none_id(self):
+        """Test is_known_issue with None issue_id."""
+        result = is_known_issue(None, 1)
+        assert result is False
+
+    def test_is_known_issue_with_none_version(self):
+        """Test is_known_issue with None issue_version."""
+        result = is_known_issue("issue123", None)
+        assert result is False
+
+    def test_is_known_issue_with_uncategorized(self):
+        """Test is_known_issue with uncategorized string."""
+        result = is_known_issue(UNCATEGORIZED_STRING, 1)
+        assert result is False
+
+    def test_is_known_issue_with_both_none(self):
+        """Test is_known_issue with both None."""
+        result = is_known_issue(None, None)
+        assert result is False
+
+
+class TestIsUnknownBuildIssue:
+    def test_is_unknown_build_issue_with_unknown_issue(self):
+        """Test is_unknown_build_issue with unknown issue."""
+        result = is_unknown_build_issue(None, None, None)
+        assert result is True
+
+    def test_is_unknown_build_issue_with_known_issue(self):
+        """Test is_unknown_build_issue with known issue."""
+        result = is_unknown_build_issue("issue123", 1, None)
+        assert result is False
+
+    def test_is_unknown_build_issue_with_test_incident(self):
+        """Test is_unknown_build_issue with test incident."""
+        result = is_unknown_build_issue(None, None, "test123")
+        assert result is False
+
+    def test_is_unknown_build_issue_with_uncategorized(self):
+        """Test is_unknown_build_issue with uncategorized."""
+        result = is_unknown_build_issue(UNCATEGORIZED_STRING, None, None)
+        assert result is True
+
+
+class TestIsExclusivelyBuildIssue:
+    def test_is_exclusively_build_issue_with_known_issue_no_test(self):
+        """Test is_exclusively_build_issue with known issue no test."""
+        result = is_exclusively_build_issue("issue123", 1, None)
+        assert result is True
+
+    def test_is_exclusively_build_issue_with_known_issue_with_test(self):
+        """Test is_exclusively_build_issue with known issue with test."""
+        result = is_exclusively_build_issue("issue123", 1, "test123")
+        assert result is False
+
```

Comment:
here isn't worth to add also a test for a unknown issue with valid test id? or its an impossible scenario?

```suggestion

    def test_is_exclusively_build_issue_with_unknown_issue_with_test(self):
        """Test is_exclusively_build_issue with unknown issue but with test."""
        result = is_exclusively_build_issue(None, None, "test123")
        assert result is False
        
```